pkg/profileparser/profileparser.go

```diff
@@ -366,6 +366,7 @@ func parseProfileFromNode(profileRoot *xmlquery.Node, pb *cmpv1alpha1.ProfileBun
 				Annotations: map[string]string{
 					cmpv1alpha1.ProductAnnotation:     productName,
 					cmpv1alpha1.ProductTypeAnnotation: string(productType),
+					cmpv1alpha1.ProfileGuidAnnotation: xccdf.GetProfileUniqueID(productName, id),
```

Comment:
A couple of thoughts here.

I think we might want to consider using a label here instead of an annotation. Mostly for two reasons.

1. The profile GUID is an identifiable attribute, and unique per profile, so we should be able to use it as a label so that users can search using it
2. The profile GUID is already a label on the `ComplianceCheckResult` and `ComplianceScan`, and letting users query those resources with a label makes it even more natural to extend to `Profile` objects.

I could see a case where someone who is looking at `ComplianceCheckResult` labels and want to use that to find the profile. With annotations, users will need to do an extra step of iterating each profile and filtering client-side by matching the label from the `ComplianceCheckResult` to the `Profile` annotations.